### DISCUSSION TITLE
Emissive Item Example

### DISCUSSION THREAD

**Discussion by TheImpalerXIII** [MoreMcmeta/core#89] [Help Wanted]
I'm new to developing mods and resource packs in general for Minecraft, and for some reason the resource pack I made just does not seem to display the emissive texture I have set.  I'm almost certain this is just because I have the resource pack set up wrong, but I can't seem to figure out the right way from the Emissive Format Guide in User Docs.  Is there any way we could have a fabric and forge demo made for this mod like we have for the base more MoreMcMeta mod?

Also, if possible, I am also curious about how to make the emissive textures dynamic similar to what is possible with predicate based overrides for textures.

**Comment by soir20** [comment]
The base demo pack does include an emissive overlay for the obsidian block; the setup is the same for items as it does for blocks. If it doesn't work, I would recommend testing without other mods because there might be an incompatibility.

Currently, there isn't support for predicate-based overrides, though you can apply animations to the emissive overlays (or make the overlay not emissive). You can create a feature request in the issues tracker, though it would be helpful to have examples of what types of predicates would be useful.

**Reply by TheImpalerXIII** [reply]
Thank you!  I had skipped over the obsidian block implementation a few times.  I was able to figure out what was going wrong with the example.  I simply didn't realize that the folder in the assets folder needed to actually match my modid.

Also apparently MoreMcMeta Emissive Textures actually does already work with predicate-based overrides if you set up the .moremcmeta file to modify the texture that is set by the override.  I attached example .json and .moremcmeta files that demonstrate how I got this to work.

[predicate_emissives.zip](https://github.com/user-attachments/files/16117799/predicate_emissives.zip)

**Reply by soir20** [reply]
Glad it was a simple fix! Yes, MoreMcmeta animations/emissives should mostly work with other mods' texture features like predicates, but MoreMcmeta doesn't add its own support for predicates.
